projects/crystal-lang.org/package.yml

```diff
@@ -0,0 +1,47 @@
+distributable:
+  url: https://github.com/crystal-lang/crystal/archive/refs/tags/{{ version }}.tar.gz
+  strip-components: 1
+
+versions:
+  github: crystal-lang/crystal
+
+# Crystal is needed to build itself…
+# So we cannot build it from source.
```

Comment:
since we're downloading a binary, can we use that to build from source ourselves?

Reply:
https://crystal-lang.org/install/from_sources/

Reply:
> since we're downloading a binary, can we use that to build from source ourselves?

But since we download the binary to compile crystal, what would be the point to download and then compile it from source?

> https://crystal-lang.org/install/from_sources/

[Install the latest Crystal release](https://crystal-lang.org/install). To compile Crystal, you need Crystal :).

Reply:
Right. That's how we build rust, and a few others I believe. They have bootstrapping built into their build process. Used to do it for Haskell but their build systems are in constant flux and it got too unstable.

In general we prefer to build our own packages, unless unavoidable.

Reply:
Ah I understand, there is a [repository](https://github.com/crystal-lang/bootstrap-script) to bootstrap the compiler but it seems pretty outdated :(.

https://github.com/crystal-lang/bootstrap-script/issues/2
https://forum.crystal-lang.org/t/segfault-while-bootstraping-from-ruby/6073/4

Reply:
Should be able to bootstrap it from itself.

Unless you want to try and get linux+aarch64 working.

Reply:
> Should be able to bootstrap it from itself.

Allright I'll try to do that when I get some free time 👍.